segmentstore/server/src/main/java/io/pravega/segmentstore/server/reading/StreamSegmentReadResult.java

```diff
@@ -29,7 +28,7 @@
     private final long streamSegmentStartOffset;
     private final int maxResultLength;
     private final NextEntrySupplier getNextItem;
-    private CompletableFuture<ReadResultEntryContents> lastEntryFuture;
+    private CompletableReadResultEntry lastEntry;
```

Comment:
What's going on with thread safety here? It should at least be documented

Reply:
Fixed the whole class to be thread safe.